apps/frontend/src/hooks/__tests__/pagination.test.ts

```diff
@@ -0,0 +1,110 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { MemoryRouter, useLocation } from 'react-router-dom';
+
+import { usePaginationParams, usePagination } from '../pagination';
+
+describe('usePaginationParams', () => {
+  it('Returns default page and page size', () => {
+    const { result } = renderHook(() => usePaginationParams(), {
+      wrapper: MemoryRouter,
+    });
+    expect(result.current.currentPage).toBe(0);
+    expect(result.current.pageSize).toBe(10);
+  });
+
+  it('Returns current page', () => {
+    const { result } = renderHook(() => usePaginationParams(), {
+      wrapper: MemoryRouter,
+      initialProps: {
+        initialEntries: [{ search: '?currentPage=1' }],
+      },
+    });
+    expect(result.current.currentPage).toBe(1);
+  });
+
+  it('Returns provided provided page size', () => {
+    const { result } = renderHook(() => usePaginationParams(12), {
+      wrapper: MemoryRouter,
+    });
+    expect(result.current.pageSize).toBe(12);
+  });
+});
+
+describe('usePagination', () => {
```

Comment:
Only case coming to my mind that is not tested is the one for `numberOfItems === 0`, can we check that the calculation works for that?